types/index.d.ts

```diff
@@ -0,0 +1,267 @@
+		include: TypedChainedSet<this, webpack.Condition>;
+		exclude: TypedChainedSet<this, webpack.Condition>;
+
+		parser(value: { [optName: string]: any }): this;
+		test(value: webpack.Condition | webpack.Condition[]): this;
+		enforce(value: 'pre' | 'post'): this;
+
+		use(name: string): Use;
+		pre(): this;
+		post(): this;
+	}
+
+	class Optimization extends ChainedMap<Config> {
+		concatenateModules(value: boolean): this;
+		flagIncludedChunks(value: boolean): this;
+		mergeDuplicateChunks(value: boolean): this;
+		minimize(value: boolean): this;
+		minimizer(name: string): Config.Plugin<this>;
+		namedChunks(value: boolean): this;
+		namedModules(value: boolean): this;
+		nodeEnv(value: boolean | string): this;
+		noEmitOnErrors(value: boolean): this;
+		occurrenceOrder(value: boolean): this;
+		portableRecords(value: boolean): this;
+		providedExports(value: boolean): this;
+		removeAvailableModules(value: boolean): this;
+		removeEmptyChunks(value: boolean): this;
+		runtimeChunk(value: boolean | "single" | "multiple" | RuntimeChunk): this;
+		sideEffects(value: boolean): this;
+		splitChunks(value: SplitChunksOptions): this;
+		usedExports(value: boolean): this;
+	}
+
+	interface RuntimeChunk {
+		name: string | RuntimeChunkFunction;
+	}
+
+	type RuntimeChunkFunction = (entryPoint: EntryPoint) => string;
+
+	interface SplitChunksOptions { [name: string]: any; }
+
+	interface LoaderOptions { [name: string]: any; }
+
+	class Use extends ChainedMap<Rule> implements Orderable {
+		loader(value: string): this;
+		options(value: LoaderOptions): this;
+
+		tap(f: (options: LoaderOptions) => LoaderOptions): this;
+
+		// Orderable
+		before(name: string): this;
+		after(name: string): this;
+	}
+
+	type DevTool = 'eval' | 'inline-source-map' | 'cheap-eval-source-map' | 'cheap-source-map' |
+		'cheap-module-eval-source-map' | 'cheap-module-source-map' | 'eval-source-map' | 'source-map' |
+		'nosources-source-map' | 'hidden-source-map' | 'nosources-source-map' | '@eval' |
+		'@inline-source-map' | '@cheap-eval-source-map' | '@cheap-source-map' |
+		'@cheap-module-eval-source-map' | '@cheap-module-source-map' | '@eval-source-map' |
+		'@source-map' | '@nosources-source-map' | '@hidden-source-map' | '@nosources-source-map' |
+		'#eval' | '#inline-source-map' | '#cheap-eval-source-map' | '#cheap-source-map' |
+		'#cheap-module-eval-source-map' | '#cheap-module-source-map' | '#eval-source-map' |
+		'#source-map' | '#nosources-source-map' | '#hidden-source-map' | '#nosources-source-map' |
+		'#@eval' | '#@inline-source-map' | '#@cheap-eval-source-map' | '#@cheap-source-map' |
+		'#@cheap-module-eval-source-map' | '#@cheap-module-source-map' | '#@eval-source-map' |
+		'#@source-map' | '#@nosources-source-map' | '#@hidden-source-map' | '#@nosources-source-map' |
+		boolean;
+
+	interface PluginClass {
+		new (...opts: any[]): webpack.Plugin;
+	}
+
+	interface Orderable {
+		before(name: string): this;
+		after(name: string): this;
+	}
+}
```

Comment:
Could you add the missing newline at the end of this file (and the others too)?

Reply:
should